src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -11041,14 +11056,34 @@ private ExpressionSyntax ParsePostFixExpression(ExpressionSyntax expr)
             }
         }
 
-        private static bool CanStartConsequenceExpression(SyntaxKind kind)
+        private bool CanStartConsequenceExpression()
         {
-            return kind is SyntaxKind.DotToken or SyntaxKind.OpenBracketToken;
+            Debug.Assert(this.CurrentToken.Kind == SyntaxKind.QuestionToken);
+            var nextToken = this.PeekToken(1);
+
+            // ?. is always the start of of a consequence expression.
+            if (nextToken.Kind == SyntaxKind.DotToken)
+                return true;
+
+            if (nextToken.Kind == SyntaxKind.OpenBracketToken)
+            {
+                // could simply be `x?[0]`, or could be `x ? [0] : [1]`.
+                using var _ = GetDisposableResetPoint(resetOnDispose: true);
+
+                var collectionExpression = this.ParseCollectionCreationExpression();
+
+                // PROTOTYPE: Def back compat concern here.  What if the user has `x ? y?[0] : z` this would be legal,
+                // but will now change to `y?[0] : z` being a ternary.  Have to decide if this is acceptable break.
+                return this.CurrentToken.Kind != SyntaxKind.ColonToken;
```

Comment:
some interesting compat cases.  we may have to be strict and require the user to write `x ? ([1, 2, 3]) : ([4, 5, 6])`.  But that seems *Really* unfortunate.  So i'm currently allowing the nicer form of `x ? [1, 2, 3] : [4, 5, 6]`, but it may cause a compat break that LDM will have to approve.